src/Microsoft.AspNet.Routing/VirtualPathData.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Routing
+{
+    /// <summary>
+    /// Represents information about the route and virtual path that are the result of
+    /// generating a URL with the ASP.NET routing middleware.
+    /// </summary>
+    public class VirtualPathData
+    {
+        private string _virtualPath;
+        private readonly RouteValueDictionary _dataToken;
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="VirtualPathData"/> class.
+        /// </summary>
+        /// <param name="router">The object that is used to generate the URL.</param>
+        /// <param name="virtualPath">The generated URL.</param>
+        public VirtualPathData(IRouter router, string virtualPath)
+            : this(router, virtualPath, dataToken: new RouteValueDictionary())
+        {
+        }
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="VirtualPathData"/> class.
+        /// </summary>
+        /// <param name="router">The object that is used to generate the URL.</param>
+        /// <param name="virtualPath">The generated URL.</param>
+        /// <param name="dataToken">The collection of custom values.</param>
+        public VirtualPathData(IRouter router, string virtualPath, RouteValueDictionary dataToken)
+        {
+            Router = router;
+            VirtualPath = virtualPath;
+            _dataToken = dataToken;
+        }
+
+        /// <summary>
+        /// Gets the collection of custom values for the <see cref="Router"/>.
+        /// </summary>
+        public RouteValueDictionary DataTokens
```

Comment:
@harshgMSFT 
In MVC5, the datatoken is used for namespace, and area, but in MVC6, there is no support of namespace, and it uses the attribute routing for area. So, there is no actual usage in framework itself. 
I talked to @Eilon and this is mainly for end users when they want to use datatoken to generate custom URLs.